graphistry/pygraphistry.py

```diff
@@ -2446,7 +2461,140 @@ def _handle_api_response(response):
+                if not fail_silent:
+                    msg = f"Failed to get token after {repeat*wait} seconds. Please re-run the login process"
+                    if in_ipython() or in_databricks() or PyGraphistry.set_sso_opt_into_type == "display":
+                        display_message_html(f"<strong>{msg}</strong>")
+                    raise Exception(msg) 
+                else:
+                    msg_text = f'{msg_text}\nGot token'
+                    print(msg_text)
+                    return
+
+            msg_text = f'{msg_text}\nGot token'
+            print(msg_text)
+        else:
+            print('Token is valid, no waiting required.')
+
+
+    @staticmethod
+    def sso_wait_for_token_html_display(repeat: int = 20, wait: int = 5, fail_silent: bool = False):
+        """Get the JWT token for SSO login and display corresponding message in HTML
+        Get the JWT token for SSO login and display corresponding message in HTML
+        """
+        from IPython.display import display, HTML
+        if not PyGraphistry.api_token():
+            msg_html = '<br /><strong> .... </strong>'
+            if not PyGraphistry.sso_repeat_get_token(repeat, wait):
+                msg_html = f'{msg_html}<br /><strong>Failed to get token after {repeat*wait} seconds .... </strong>'
+                if not fail_silent:
+                    raise Exception(f"Failed to get token after {repeat*wait} seconds. Please re-run the login process") 
+                else:
+                    msg_html = f'{msg_html}<br /><strong>Got token</strong>'
+                    display(HTML(msg_html))
+                    return
+
+            msg_html = f'{msg_html}<br /><strong>Got token</strong>'
+            display(HTML(msg_html))
+        else:
+            display(HTML('<br /><strong>Token is valid, no waiting required.</strong>'))
+
+
+    @staticmethod
+    def sso_verify_token_display(
+        repeat: int = 20,
+        wait: int = 5,
+        display_mode: str = 'text'
+    ) -> bool:
+        if display_mode == 'html':
+            from IPython.display import display, HTML, clear_output
+            clear_output()
+
+        required_login = False
+        token = PyGraphistry.api_token()
+        if token:
+            is_valid = PyGraphistry.verify_token()
+            print(f"is_valid : {is_valid}")
+            if not is_valid:
+                print("***********token not valid, refresh token*****************")
+                if display_mode == 'html':
+                    display(HTML('<br /><strong>Refresh token ....</strong>'))
+                try:
+                    PyGraphistry.refresh()
+                except Exception:
+                    required_login = True
+
+            else:
+                print("Token is still valid")
+                if display_mode == 'html':
+                    display(HTML('<br /><strong>Token is still valid ....</strong>'))
```

Comment:
`display(HTML('<br /><strong>Token is still valid; no re-login required.</strong>'))
`